py/tests/core/test_config_schema.py

```diff
@@ -0,0 +1,77 @@
+from configu import ConfigSchema
+from configu.core.config_schema import ConfigSchemaDefinition
+import pytest
+
+@pytest.mark.parametrize("path", (
+  "tests/staticfiles/config_schemas/valid/empty.cfgu.json",
+  "tests/staticfiles/config_schemas/valid/basic.cfgu.json",
+  "tests/staticfiles/config_schemas/valid/complex.cfgu.json",
+))
+def test_valid_config_schemas(path):
+    ConfigSchema.parse(ConfigSchema(path))
+
+@pytest.mark.parametrize("path", (
+    "tests/staticfiles/config_schemas/invalid/regex_missing_pattern.cfgu.json",
+    "tests/staticfiles/config_schemas/invalid/reserved_names.cfgu.json",
+    "tests/staticfiles/config_schemas/invalid/default_and_required.cfgu.json",
+    "tests/staticfiles/config_schemas/invalid/default_and_template.cfgu.json",
+    "tests/staticfiles/config_schemas/invalid/nonexistent_type.cfgu.json",
+    "tests/staticfiles/config_schemas/invalid/invalid_default.cfgu.json",
+))
+def test_invalid_config_schemas(path):
+    with pytest.raises(ValueError):
+        ConfigSchema.parse(ConfigSchema(path))
+
+@pytest.mark.parametrize("path", (
+    # "tests/staticfiles/config_schemas/invalid/bad_extension.json", // TODO: This currrent fails. Is this a bug?
+    "tests/staticfiles/config_schemas/invalid/bad_file_type.txt",
+))
+def test_invalid_extension(path):
+    with pytest.raises(ValueError):
+      ConfigSchema(path)
+
+@pytest.mark.parametrize("validator,value,expected", (
+  ("Boolean", "True", True),
+  ("Boolean", "Foo", False),
+  ("Number", "3.14159", True),
+  ("Number", "asd", False),
+  ("String", "Hello", True),
+  ("UUID", "0d3d45c8-11da-4730-8e90-30c2e04d51ec", True),
+  ("UUID", "Foo", False),
+  ("SemVer", "3.2.2", True),
+  ("SemVer", "not.a.semver", False),
+  ("Email", "test@example.com", True),
+  ("Email", "not-an-email", False),
+  ("MobilePhone", "+1 (200) 200 0000", True),
+  ("MobilePhone", "not-a-mobile-number", False),
+  ("LatLong", "22.2636563,-49.1842674", True),
+  ("LatLong", "not-a-lat-long", False),
+  ("Color", "ffffff", True),
+  ("Color", "not-a-color", False),
+  ("IPv4", "192.168.0.0", True),
+  ("IPv4", "not-an-ipv4", False),
+  ("IPv6", "0000:0000:0000:0000:0000:0000:0000:0001", True),
+  ("IPv6", "not-an-ipv6", False),
+  ("Domain", "example.com", True),
+  ("Domain", "not-a-domain", False),
+  ("URL", "https://example.com", True),
+  ("URL", "not-a-url", False),
+  ("ConnectionString", "user@service/path?opts", True),
+  # ("ConnectionString", "", False), # TODO: Couldn't find any string that doesn't match the regex.
```

Comment:
@RonConfigu any ideas on this? As far as I can tell it'll match literally any string, in which case the regex may be too broad? Alternatively if you know of a string that doesn't match the regex please let me know and I'll use it for this test.

Reply:
We'll wait for Ran with this one. The question is in the RegEx itself and it maybe exactly what he want it to be.
Regardless. Good catch! 👍

Reply:
:+1: In that case I'll leave it commented out with a TODO comment for now, to be tackled once a decision has been made regarding the regex.

Reply:
Opened issue regarding this.
https://github.com/configu/configu/issues/233
Thanks @jamesL92